### PR TITLE
Include annotation values in scale min/max range

### LINKED ISSUE #23: Chart scales are not adjusted with annotations
The automatic scale does not account for the annotations. If I set an annotation with value greater than the data scale the annotation is drawn outside the chart area.

### PR DESCRIPTION
**Pull request by compwright**
Resolves #23

### CODE REVIEW

**Comment by compwright** [comment]
@etimberg Could you look over this and give me your feedback? It does have a bug that I can't seem to find - whenever a scale min or max is changed, a line annotation will not show up. If another line is added that does not require a scale adjustment, both lines then appear.

You can see this behavior in the updates.html sample.

File: src/plugin.js [hunk 1]
```diff
@@ -80,19 +80,74 @@ function buildAnnotations(configs) {
...
+				var scale = chartInstance.scales[scaleId];
+				decorate(scale, 'determineDataLimits', function(previous) {
+					previous();
+					var dataMin = scale.min;
+					var dataMax = scale.max;
+					var range = getAnnotationDataLimits(scaleId, chartInstance.annotations, scale.min, scale.max);
+					scale.min = range.min;
+					scale.max = range.max;
+					if (scale.min < dataMin) {
+						console.log('Adjusted', scaleId, 'min from', dataMin, 'to', scale.min);
```
Comment (etimberg) [inline src/plugin.js]: might not be a good idea to log this in production.

File: src/types/box.js [hunk 2]
```diff
@@ -21,20 +23,31 @@ module.exports = function(Chart) {
...
 			if (xScale) {
-				min = isValid(options.xMin) ? xScale.getPixelForValue(options.xMin) : chartArea.left;
-				max = isValid(options.xMax) ? xScale.getPixelForValue(options.xMax) : chartArea.right;
+				min = helpers.isValid(options.xMin) ? xScale.getPixelForValue(options.xMin) : chartArea.left;
+				max = helpers.isValid(options.xMax) ? xScale.getPixelForValue(options.xMax) : chartArea.right;
 				left = Math.min(min, max);
 				right = Math.max(min, max);
+
+				model.ranges[options.xScaleID] = {
+					min: xScale.getValueForPixel(left),
```
Comment (etimberg) [inline src/types/box.js]: I was going to comment that you could use `xScale.min` and `xScale.max` but I realized I don't think we explicitly make those public so good call doing it this way

File: src/plugin.js (hunk 1 above)
Comment (compwright) [inline src/plugin.js]: It won't be, this is temporary.

**Comment by etimberg** [comment]
@compwright i noticed that bug during update. I think you need to `decorate` in the `beforeUpdate` plugin call so that the axis updates with the correct sizes but actually configuring the annotations needs to happen in `afterUpdate` otherwise the pixel values will be wrong.

**Pull request merged**